src/main/java/lotto/domain/RankStatistic.java

```diff
@@ -0,0 +1,53 @@
+package lotto.domain;
+
+import static lotto.domain.LottoGames.LOTTO_PRICE;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lotto.domain.model.Rank;
+
+public class RankStatistic {
+
+  private Map<Rank, Long> rankStatistic;
+
+  public RankStatistic(List<Rank> ranks) {
+    rankStatistic = ranks.stream()
+        .collect(
+            Collectors.groupingBy(
+                Function.identity(),
+                Collectors.counting()
+            )
+        );
+  }
+
+  public double calculateYield() {
+    return calculatePrizeMoney() / (double) calculatePurchaseAmount();
+  }
+
+  private long calculatePrizeMoney() {
+    return rankStatistic.entrySet()
+        .stream()
+        .mapToLong(entry -> {
+          Rank rank = entry.getKey();
+          long winCount = entry.getValue();
+          return rank.getWinningMoney() * winCount;
+        })
+        .sum();
+  }
+
+  private long calculatePurchaseAmount() {
+    return rankStatistic.values()
+        .stream()
+        .mapToLong(v -> v)
+        .sum() * LOTTO_PRICE;
```

Comment:
```suggestion
                .reduce((long) 0, (a, b) -> a + b)
                * LOTTO_PRICE;
```

어차피 LongStream인데요, boxing을 하기보다는 reduce 를 활용하는 것은 어떨가요